core/src/upickle/core/Types.scala

```diff
@@ -184,18 +184,16 @@ trait Types{ types =>
     }
   }
 
-  abstract class CaseR[V](val argCount: Int) extends SimpleReader[V]{
+  abstract class CaseR[V] extends SimpleReader[V]{
     override def expectedMsg = "expected dictionary"
     trait CaseObjectContext extends ObjVisitor[Any, V]{
-      val aggregated = new Array[Any](argCount)
-      val found = new Array[Boolean](argCount)
+      def storeAggregatedValue(currentIndex: Int, v: Any): Unit
+      var found = 0L
```

Comment:
Now that we have a field limit of 64, dropping this from `long` to `byte` might make the `CaseObjectContext` fit into a smaller 8-byte alignment on the jvm--particularly alongside the new potentially primitive-typed vars.

Reply:
Oops, this is the bitset. Meant to post this on `currentIndex`.